libs/ssl-config/src/test/java/org/elasticsearch/common/ssl/SslDiagnosticsTests.java

```diff
@@ -126,7 +126,8 @@ public void testDiagnosticMessageWhenServerProvidesEndCertificateOnlyButTheCertA
             " the certificate has subject alternative names [DNS:localhost,IP:127.0.0.1];" +
             " the certificate is issued by [CN=Test CA 1]" +
             " but the server did not provide a copy of the issuing certificate in the certificate chain;" +
-            " this ssl context ([xpack.http.ssl]) is not configured to trust that issuer"));
+            " this ssl context ([xpack.http.ssl]) is not configured to trust that issuer" +
+            " but trusts [2] other issuers ([CN=Test CA 2, CN=Test CA 3])"));
```

Comment:
For completeness, should we have a test where the number is greater than 9?

Reply:
I will add that.
I have a philosophical question about it - if I had included this part of the message in the original PR to add diagnostics, I don't think anyone would have expected it to have a test to cover every single case, but we tend to lean that way when we add small changes after the fact.

I wonder whether that means we should be aiming for smaller PRs in general, or most extensive testing on the larger PRs. Our current approach of doing more complete testing on the followup PRs than we do on the first delivery doesn't seem right.

Reply:
I'd personally agree it has something to do with the size of the PR. Large PRs are hard to be exhaustive. They also tend to run for longer and hence people are more easily to get less focused. On the other hand, smaller PRs are easier to get nitpicked since there aren't many things to talk about. It's a different level of zooming due to the PR sizes( and possibly human nature?). In an ideal world, I'd prefer every PR to be small. But that's pratically impossible. Overall, I don't think what we are doing is perfect, but it is not seriously flawed either.